### PR TITLE
feat: create/consume branch flag to filter PVL

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
adds a `--branch` filter to `package version list` command

will only list packages on the specified branch

### What issues does this PR fix or reference?
@W-11167396@
REQUIRES: https://github.com/forcedotcom/packaging/pull/594

### CODE REVIEW

File: messages/package_version_list.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ All filter parameters are applied using the AND logical operator (not OR).
 
 Display limited package version details.
 
+# flags.branch.summary
+
+Use to filter the results to a specific branch
```
Comment (jshackell-sfdc) [inline messages/package_version_list.md:17]:
```suggestion
Branch in your source control system used to filter the results; only package versions based on the specified branch are listed. 
```

Comment (jshackell-sfdc) [reply messages/package_version_list.md:17]: @WillieRuemmele , this is a bit wordy, I know.  But stylistically I can't start the description with a verb, I need to start with a description of the flag's value.  Let me know what you think, I can try again if you don't like it.

Comment (jshackell-sfdc) [reply messages/package_version_list.md:17]: @WillieRuemmele , how about adding this new flag to an example, maybe the fourth one?  So it would be:

- List package versions that are based on the "featureA" branch in your source control system that were modified today in your default Dev Hub org; show limited details about each one:

**Review by jshackell-sfdc — CHANGES_REQUESTED** [review]
see my suggestions

**Review by shetzel — APPROVED** [review]

**Review by jshackell-sfdc — APPROVED** [review]
